lib/irb.rb

```diff
@@ -21,36 +21,917 @@
+#
+# - The value of variable <tt>$IRBRC</tt>, if defined.
+# - The value of variable <tt>$XDG_CONFIG_HOME/irb/irbrc</tt>, if defined.
+# - File +.config/irb/irbrc+, if it exists.
+# - File +.irbrc+, if it exists.
+# - File +irb.rc+, if it exists.
+# - File +_irbrc+, if it exists.
+# - The value of variable <tt>$irbrc</tt>, if defined.
+#
+# If the search fails, there is no configuration file.
+#
+# If the search succeeds, the configuration file is read as Ruby code,
+# and so can contain any Ruby programming you like.
+#
+# \Method <tt>conf.rc?</tt> returns +true+ if a configuration file was read,
+# +false+ otherwise.
+# \Hash entry <tt>IRB.conf[:RC]</tt> also contains that value.
+#
+# [REVIEWERS] What should be said about IRB.conf[:RC_NAME_GENERATOR]?
+#
+# === Session \Context
+#
+# An \IRB session has a <i>context</i>, an IRB::Context object
+# that is created at session startup.
+# Its initial content is determined by command-line options,
+# by entries in hash <tt>IRB.conf</tt>,
+# and by defaults.
+#
+# Values in the context can be retrieved
+# via methods and attributes in the context object;
+# much of its content can be modified by other methods and attributes.
+#
+# \Hash entry <tt>IRB.conf[:MAIN_CONTEXT]</tt>
+# contains the context for the main session.
+#
+# === \Hash <tt>IRB.conf</tt>
+#
+# The initial entries in hash <tt>IRB.conf</tt> are determined by:
+#
+# - Default values.
+# - Command-line options, which may override defaults.
+# - Direct assignments in the configuration file.
+#
+# You can see the hash by typing <tt>IRB.conf</tt>.
+#
+# Details of the entries' meanings are described in the relevant subsections below.
+#
+# If you are interested in a specific entry, consult the
+# {index}[./doc/irb/indexes_rdoc.html#label-Index+of+IRB.conf+Entries].
+#
+# === Notes on Initialization Precedence
+#
+# - Any conflict between an entry in hash <tt>IRB.conf</tt> and a command-line option
+#   is resolved in favor of the hash entry.
+# - \Hash <tt>IRB.conf</tt> affects the context only once,
+#   when the configuration file is interpreted;
+#   any subsequent changes to it do not affect the context
+#   and are therefore essentially meaningless.
+#
+# === Initialization Script
+#
+# By default, the first command-line argument (after any options)
+# is the path to a Ruby initialization script.
+#
+# \IRB reads the initialization script and puts its content onto the \IRB shell,
+# just as if it were user-typed commands.
+#
+# Command-line option <tt>--noscript</tt> causes the first command-line argument
+# to be treated as an ordinary argument (instead of an initialization script);
+# <tt>--script</tt> is the default.
+#
+# == Input
+#
+# This section describes the features that allow you to change
+# the way \IRB input works;
+# see also {Input and Output}[rdoc-ref:IRB@Input+and+Output].
+#
+# === Input Command History
+#
+# By default, \IRB stores a history of up to 1000 input commands
+# in file <tt>./.irb_history</tt>.
```

Comment:
I'm pretty sure it stores it in the home directory rather than the current directory.

```suggestion
# in file <tt>~/.irb_history</tt>.
```

Reply:
```$ ls -l .irb_history
-rw------- 1 burdette burdette 1205 Oct 24 11:12 .irb_history
$ irb
irb(main):001> 1 + 1
=> 2
irb(main):002> exit
$ ls -l .irb_history
-rw------- 1 burdette burdette 1216 Oct 24 11:31 .irb_history
```

Reply:
/home/burdette/Documents/GitHub/irb

Reply:
That's very odd. What if you deleted the `.irb_history` file in the currently directory? Does it create that file and append to it?

Reply:
```
$ rm .irb_history
$ irb
irb(main):001> 1 + 1
=> 2
irb(main):002> exit
$ ls -l .irb_history
-rw------- 1 burdette burdette 11 Nov  4 12:53 .irb_history
$ cat .irbrc

$ 
```

Reply:
IRB will save `.irb_history` to the same directory of `.irbrc`.
```
$ mkdir tmpdir
$ touch tmpdir/.irbrc
$ IRBRC=tmpdir/.irbrc irb
irb(main):001> irb_info
......
.irbrc path: tmpdir/.irbrc
......
irb(main):002> exit
$ ls -a tmpdir
./              ../             .irbrc          .irbrc_history
```

Normally, irb will save to `~/.irb_history`. If `.irbrc` exists in current directory, then history will saved to current directory.

Reply:
Ah, thanks much!

Reply:
@tompng, would it be correct to say that if any IRB resource file is found, the history file will be in that same directory?

Reply:
@tompng, I have changed the text at l. 202.

Reply:
Changed accordingly.